components/order-app/stores/order-utils.ts

```diff
@@ -20,13 +25,16 @@ export const distanceFromTMS = async (googlePlacesObj: google.maps.places.PlaceR
   await insertGoogleMapsScript();
 
   const TMSLocation = new google.maps.LatLng(28.539307, -81.286839);
-  const result = await getDistance(TMSLocation, googlePlacesObj.geometry.location);
-  const distanceInMiles = Number(result.rows[0].elements[0].distance.text.split(' ')[0]);
-  return distanceInMiles;
+  if (googlePlacesObj.geometry && googlePlacesObj.geometry.location) {
+    const result = await getDistance(TMSLocation, googlePlacesObj.geometry.location);
+    const distanceInMiles = Number(result.rows[0].elements[0].distance.text.split(' ')[0]);
+    return distanceInMiles;
+  }
+  return 0;
 };
 
 export const formatGooglePlacesObj = ({ name, formatted_address }: google.maps.places.PlaceResult) => {
-  if (formatted_address.includes(name)) {
+  if (formatted_address && name && formatted_address.includes(name)) {
```

Comment:
If these are required parameters, why do we need to check for them?

Reply:
Actually, `name` and `formatted_address` can be undefined. Is this solution fine with you? With this solution, we are telling the typescript compiler that `name` and `formatted_address` won't be undefined or null.

```ts
export const formatGooglePlacesObj = ({
  name,
  formatted_address,
}: Required<Pick<google.maps.places.PlaceResult, 'name' | 'formatted_address'>>) => {
  if (formatted_address.includes(name)) {
    return formatted_address;
  } else {
    return `${name}, ${formatted_address}`;
  }
};
```

Reply:
Is this solution fine with you?

```ts
export const formatGooglePlacesObj = (googlePlacesObj: google.maps.places.PlaceResult) => {
  const name = googlePlacesObj.name!;
  const formatted_address = googlePlacesObj.formatted_address!;
  if (formatted_address.includes(name)) {
    return formatted_address;
  } else {
    return `${name}, ${formatted_address}`;
  }
};
```

Reply:
No, just keep as you had it before then